### PR TITLE
Add the UA slot 'initiators of active Picture-in-Picture sessions'

### PR DESCRIPTION
**Pull request by kenchris**
Add the internal UA slot 'initiators of active Picture-in-Picture sessions'

This allows other specs, like Compute Pressure, to know if a document
currently has an active Picture-in-Picture session.

This is related to https://github.com/WICG/compute-pressure/issues/83

***
<a href="https://pr-preview.s3.amazonaws.com/kenchris/picture-in-picture/pull/210.html" title="Last updated on Mar 22, 2022, 8:28 AM UTC (4349ce6)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/picture-in-picture/210/fb22039...kenchris:4349ce6.html" title="Last updated on Mar 22, 2022, 8:28 AM UTC (4349ce6)">Diff</a>

### CODE REVIEW

**Comment by kenchris** [comment]
@beaufortfrancois fyi

**Comment by kenchris** [comment]
> Thanks for starting this PR @kenchris!
> Can you share an example of how you'll use it in Compute Pressure API spec for instance?

This will be used in https://wicg.github.io/compute-pressure/#supporting-algorithms

> If [=user agent=]'s [=initiators of active Picture-in-Picture sessions=] [=set/contains=] o's [associated document](https://html.spec.whatwg.org/multipage/window-object.html#concept-document-window), return true.

File: index.bs [hunk 1]
```diff
@@ -179,29 +186,32 @@ run the following steps:
...
-8. If |playingRequired| is `true` and |video| is {{paused}}, abort these steps.
-9. Set {{pictureInPictureElement}} to |video|.
-10. Let <dfn>Picture-in-Picture window</dfn> be a new instance of
+8. If |video| is {{pictureInPictureElement}}, abort these steps.
+9. If |playingRequired| is `true` and |video| is {{paused}}, abort these steps.
+10. Set {{pictureInPictureElement}} to |video|.
+11. Let <dfn>Picture-in-Picture window</dfn> be a new instance of
     {{PictureInPictureWindow}} associated with {{pictureInPictureElement}}.
-11. <a>Queue a task</a> to <a>fire an event</a> with the name
+12. Append |document| to <a>initiators of active Picture-in-Picture sessions</a>.
```
[the remark this answers is not in the document]
Comment (kenchris) [reply index.bs:208]: I could, but then I would need to get the document from it and handle cases like it being detached

[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply index.bs:208]: Got it.

**Review by beaufortfrancois — APPROVED** [review]
LGTM

@jernoble Does that look good to you as well?

**Comment by kenchris** [comment]
Thanks @beaufortfrancois - I see that I need to change this to store origin instead, as I need to use this from workers (incl. shared workers). I will update the PR

File: index.bs [hunk 2]
```diff
@@ -179,29 +186,33 @@ run the following steps:
 
 1. If <a>Picture-in-Picture support</a> is `false`, throw a
     {{NotSupportedError}} and abort these steps.
-2. If the document is not <a>allowed to use</a> the <a>policy-controlled feature</a>
+2. Let |document| be the <a>current settings object</a>'s <a>relevant global object</a>'s
```
Comment (beaufortfrancois) [inline index.bs:189]: Then you can revert this change so that your PR affects only the last part of this algorithm

Comment (kenchris) [reply index.bs:189]: I can but you want that change in another PR then? Otherwise it is not clear where document comes from

Comment (kenchris) [reply index.bs]: Done!

**Comment by kenchris** [comment]
Friendly ping (❁´◡`❁)

**Comment by beaufortfrancois** [comment]
I'll use my editor hat to merge this CL.
@jernoble Let us know if we've missed something. We can always update the spec later.

**Comment by kenchris** [comment]
Thank you!
